src/Groups/homomorphisms.jl

```diff
@@ -0,0 +1,360 @@
+  H1 = GAP.Globals.Image(f.map, H.X)
+  return _as_subgroup(codomain(f), H1)
+end
+
+"""
+    cokernel(f::GAPGroupHomomorphism)
+Return the cokernel of `f`.
+"""
+function cokernel(f::GAPGroupHomomorphism)
+  K, mK = image(f)
+  return quo(codomain(f), K)
+end
+
+"""
+TODO: document this
+"""
+function haspreimage(f::GAPGroupHomomorphism, x::GAPGroupElem)
+  r = GAP.Globals.PreImagesRepresentative(f.map, x.X)
+  if r == GAP.Globals.fail
+    return false, one(domain(f))
+  else
+    return true, group_element(domain(f), r)
+  end
+end
+
+#TODO fix this
+"""
+    preimage(f::GAPGroupHomomorphism{S, T}, H::T) where S <: GAPGroup where T <: GAPGroup
+If `H` is a subgroup of the codomain of `f`, return the subgroup `f^-1(H)`, together with its embedding homomorphism into the domain of `f`.
+"""
+function preimage(f::GAPGroupHomomorphism{S, T}, H::T) where S <: GAPGroup where T <: GAPGroup
+  H1 = GAP.Globals.PreImage(f.map, H.X)
+  return _as_subgroup(domain(f), H1)
+end
+
+
+################################################################################
+#
+#  IsIsomorphic
+#
+################################################################################
+
+"""
+    isisomorphic(G::Group, H::Group)
+Return (`true`,`f`) if `G` and `H` are isomorphic groups, where `f` is a group isomorphism. Otherwise, return (`false`,`f`), where `f` is the trivial homomorphism.
+"""
+function isisomorphic(G::GAPGroup, H::GAPGroup)
+  mp = GAP.Globals.IsomorphismGroups(G.X, H.X)
+  if mp == GAP.Globals.fail
+    return false, trivial_morphism(G, H)
+  else
+    return true, _hom_from_gap_map(G, H, mp)
+  end
+end
+
+
+################################################################################
+#
+#  Automorphism Group
+#
+################################################################################
+
+"""
+    automorphism_group(G::Group) -> A::AutomorphismGroup{T}
+Return the full automorphism group of `G`. If `f` is an object of type ``GAPGroupHomomorphism`` and it is bijective from `G` to itself, then `A(f)` return the embedding of `f` in `A`. 
+
+Elements of `A` can be multiplied with other elements of `A` or by elements of type ``GAPGroupHomomorphism``; in this last case, the result has type ``GAPGroupHomomorphism``.
+"""
+function automorphism_group(G::GAPGroup)
+  AutGAP = GAP.Globals.AutomorphismGroup(G.X)
+  return AutomorphismGroup{typeof(G)}(AutGAP, G)
+end
+
+function Base.show(io::IO, A::AutomorphismGroup{T}) where T <: GAPGroup
+  print(io, "Aut( "* GAP.gap_to_julia(GAP.Globals.StringView(A.G.X)) *" )")
+end
+
+# TODO: why (x::AutomorphismGroupElem) does not work?
+# Answer: because AutomorphismGroupElem = GAPGroupElem{AutomorphismGroup} is NOT a sup-type of GAPGroupElem{AutomorphismGroup{T}} for any T
+
+"""
+    hom(f::GAPGroupElem{AutomorphismGroup{T}}) where T
+Return the element f of type ``GAPGroupHomomorphism{T,T}``.
+"""
+function hom(x::AutomorphismGroupElem)
```

Comment:
I am confused by this comment, docstring and function -- the comment says why the function won't work (I think ?!?) and the docstring shows how it should look like instead?

Is there  test for this `hom` method? If this docstring contained a `jldoctest` (as explained in another comment), I wouldn't have to ask :-)